galaxy_utils/sequence/scripts/fastq_groomer.py

```diff
@@ -11,51 +11,89 @@
 )
 
 
+class Groomer():
+
+    def __init__(self, file_handle=None):
+        self.file_handle = file_handle
+        args = self._get_args()
+        self.input_filename = args.input_filename
+        self.input_type = args.input_type
+        self.output_filename = args.output_filename
+        self.output_type = args.output_type
+        self.force_quality_encoding = args.force_quality_encoding
+        self.summarize_input = args.summarize_input == 'summarize_input'
+        self.fix_id = args.fix_id
+
+        if self.force_quality_encoding == 'None':
+            self.force_quality_encoding = None
+
+    def run(self):
+        aggregator = fastqAggregator()
+        out = fastqWriter(
```

Comment:
Could you turn `fastqWriter` and `fastqReader` into context managers by adding `__enter__` and `__exit__` and then use `with fastqReader(**kwargs) as input, fastqWriter(**kwargs) as writer:` ? Otherwise you should you should probably use a try/finally to close the files at the end.

Reply:
Possibly. I'm rereading my code (it's been months). Will respond later today.

Reply:
@mvdbeek : all done. I've converted `fastqReader` and `fastqWriter` to context managers. These are used in all `sequence/scripts` - so that led to more edits. It is my understanding that these classes are *not* used outside this suite. If they are (i.e., if we have code accessing these 2 classes directly), this is a breaking change. Both classes opened the file in their constructors; now this happens in the `__enter__` method: so the file is opened only when the `with` statement is evaluated. But again, as far as I know, the reader/writer are *not* used by outside code.

Implementation detail: I moved some local variables outside the `with` statement if they were accessed *after* that statement, like so:
```
i = None  # in the original code, this was assigned after the reader
with reader:
  i += 1
print(i)
```
I know `with` doesn't define its own scope, but in the unlikely event that (a) an exception is raised and (b) the `__exit__` method has been updated to handle that exception, the `print(i)` line would raise an error. Unlikely scenario, but that's keeping it safe.

Also, I did not convert other readers/writers (like the ones in `fasta.py`) because there are no tests for those.. Also, there are at least a few other places in the code where file handles may remain open. But that would be a different PR.